versions/3.1.0.md

```diff
@@ -2675,9 +2674,14 @@ components:
 
 #### <a name="discriminatorObject"></a>Discriminator Object
 
-When request bodies or response payloads may be one of a number of different schemas, a `discriminator` object can be used to aid in serialization, deserialization, and validation.  The discriminator is a specific object in a schema which is used to inform the consumer of the document of an alternative schema based on the value associated with it.
+When request bodies or response payloads may be one of a number of different schemas, a `discriminator` object can be used to aid in serialization, deserialization, and validation. The discriminator is a specific object in a schema which is used to inform the consumer of the document of an alternative schema based on the value associated with it.
 
-When using the discriminator, _inline_ schemas will not be considered.
+A discriminator MAY act as a "hint" to shortcut validation and selection of the matching schema which may be a costly operation, depending on the complexity of the schema.
```

Comment:
I moved this sentence up in the section because it was in the middle of an example, but this sentence is always true regardless of the specific example that was provided (?).

What does the word MAY mean for validation exactly? Who decides whether the discriminator is used as a hint or not? It would be good to be completely unambiguous such that there is no discrepancy between OAS validators.

1. Loop through all _inline_ child schemas and apply JSON schema validation (because discriminators are not "considered" for inline schemas?)
1. Loop through all _referenced_ child schema and use discriminator as a hint to determine if payload matches child schema or not. Then is the validator supposed to validate the matching payload against the JSON schema. I would think the JSON schema should still be applied after the discriminator match, but the sentence seem to imply validation may be skipped.